src/Schema/Directives/RelationDirective.php

```diff
@@ -54,15 +55,16 @@ public function resolveField(FieldValue $fieldValue): callable
             // that we know to be present on the parent model.
             if (
                 $this->lighthouseConfig['shortcut_foreign_key_selection']
-                && (array_diff_assoc($resolveInfo->getFieldSelection(), ['id' => true, '__typename' => true]) === [])
+                && (array_diff(array_keys($resolveInfo->getFieldSelection()), ['id', '__typename']) === [])
```

Comment:
Can you show me which test case is failing when reverting this line, and how?

Reply:
Got it from https://github.com/nuwave/lighthouse/actions/runs/13946583856/job/39035112800, thanks.